nautobot/utilities/templatetags/helpers.py

```diff
@@ -681,3 +681,16 @@ def custom_branding_or_static(branding_asset, static_asset):
     if settings.BRANDING_FILEPATHS.get(branding_asset):
         return f"{ settings.MEDIA_URL }{ settings.BRANDING_FILEPATHS.get(branding_asset) }"
     return StaticNode.handle_simple(static_asset)
+
+
+@library.filter()
+@register.filter()
+def render_obj_display(model, field_name):
+    """Render the display view of an object."""
+    obj = getattr(model, field_name)
+    if obj:
+        color = getattr(obj, "color", None)
+        display = getattr(model, f"get_{field_name}_display")
+        content = f'<span class="label" style="color: {fgcolor(color)}; background-color: #{color}">{display()}</span>'
+        return mark_safe(content)
```

Comment:
Please test this against Status/Role objects that have a `name` containing HTML-unsafe characters like `<` and `>` - I suspect we have an issue here. Probably should be using `format_html()` here.